BusinessLogic/Entities/Auth/OAuthClientPassword.cs

```diff
@@ -117,15 +117,30 @@ public async Task<bool> SendEvent(Event e, Subscription subscription)
         /// <param name="response"></param>
         private void ResetAccessTokenIfErrorInResponseMessage(HttpResponseMessage response)
         {
-            List<Dictionary<string, string>> resultValues = JsonConvert.DeserializeObject<List<Dictionary<string, string>>>(response.Content.ReadAsStringAsync().Result);
 
-            if (resultValues.Count > 0)
+            // if error code string is there, then parse
+            if (response.Content.ReadAsStringAsync().Result.IndexOf("errorCode") != -1)
             {
-                Dictionary<string, string> firstItem = resultValues[0];
-                if (firstItem.ContainsKey("errorCode") && firstItem["errorCode"] == "INVALID_SESSION_ID")
+                // try to parse the json to find the error code
+                try
                 {
-                    Log.Debug($"OAuthClientPassword.ResetAccessTokenIfErrorInResponseMessage: AccessToken set to `null` ");
-                    AccessToken = null;
+                    List<Dictionary<string, string>> resultValues = JsonConvert.DeserializeObject<List<Dictionary<string, string>>>(response.Content.ReadAsStringAsync().Result);
```

Comment:
```suggestion
                    List<Dictionary<string, string>> resultValues = JsonConvert.DeserializeObject<List<Dictionary<string, string>>>(await response.Content.ReadAsStringAsync());
```
same suggestion here regarding the `.Result`

Reply:
same, the await is on line 92